logic/movies.go

```diff
@@ -0,0 +1,436 @@
+package logic
+
+import (
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/zorchenhimer/MoviePolls/models"
+)
+
+var re_tagSearch = regexp.MustCompile(`t:"([a-zA-Z ]+)"`)
+
+func (b *backend) SearchMovieTitles(query string) ([]*models.Movie, error) {
+	// finding tags
+	tags := re_tagSearch.FindAllString(query, -1)
+
+	// clean up the tags from the "tagsyntax"
+	tagsToFind := []string{}
+	for _, tag := range tags {
+		tagsToFind = append(tagsToFind, tag[3:len(tag)-1])
+	}
+
+	query = re_tagSearch.ReplaceAllString(query, "")
+	query = strings.Trim(query, " ")
+
+	// we first seach for matching titles (ignoring the tags for now)
+	movieList, err := b.data.SearchMovieTitles(query)
+	if err != nil {
+		return nil, err
+	}
+
+	// NOW we filter the already found movies by the tags provided
+	return models.FilterMoviesByTags(movieList, tagsToFind)
+}
+
+func (b *backend) GetActiveMovies() ([]*models.Movie, error) {
+	return b.data.GetActiveMovies()
+}
+
+func (b *backend) GetMovie(id int) *models.Movie {
+	m, err := b.data.GetMovie(id)
+	if err != nil {
+		b.l.Error("Error getting movie with ID %d: %v", id, err)
+		return nil
+	}
+	return m
+}
+
+func (b *backend) validateForm(fields map[string]*InputField) (map[string]*InputField, bool, []*models.Link) {
+
+	ret, links := b.parseLinks(fields["Links"])
+
+	if ret.Error != nil {
+		return fields, false, nil
+	}
+
+	fields["Links"] = ret
+
+	// Check Remarks max length
+	maxRemarksLength, err := b.GetMaxRemarksLength()
+	if err != nil {
+		b.l.Debug("%v", err.Error)
+		fields["Remarks"].Error = fmt.Errorf("Something went wrong :C")
+		return fields, false, nil
+	}
+
+	length := models.GetStringLength(fields["Remarks"].Value)
```

Comment:
Why is this in models

Reply:
this one is not that easy to fix, since if it were in the backend we would have an import cycle, maybe we want to do a "common" package for utility stuff used everywhere anyways? @zorchenhimer thoughts?